core/src/test/java/tech/pegasys/web3signer/core/service/Eth1AddressSignerIdentifierTest.java

```diff
@@ -15,53 +15,89 @@
+    final String mixCaseAddress = "0X" + convertAlphabetsToUpperCase(address);
+
+    final Eth1AddressSignerIdentifier signerIdentifier =
+        new Eth1AddressSignerIdentifier(mixCaseAddress);
+    assertThat(signerIdentifier.toStringIdentifier()).isEqualTo(address);
+    assertThat(signerIdentifier.toStringIdentifier()).doesNotStartWithIgnoringCase("0x");
+    assertThat(signerIdentifier.toStringIdentifier()).isLowerCase();
   }
 
   @Test
   void validateWorksForSamePrimaryKey() {
-    final ECPublicKey publicKey = PublicKeyUtils.createKeyFrom("0xab");
+    final ECPublicKey publicKey = (ECPublicKey) secp256k1KeyPair.getPublic();
     final SignerIdentifier signerIdentifier = Eth1AddressSignerIdentifier.fromPublicKey(publicKey);
     assertThat(signerIdentifier.validate(publicKey)).isTrue();
   }
 
   @Test
   void validateFailsForDifferentPrimaryKey() {
-    final ECPublicKey publicKey = PublicKeyUtils.createKeyFrom("0xab");
+    final ECPublicKey publicKey = (ECPublicKey) secp256k1KeyPair.getPublic();
     final SignerIdentifier signerIdentifier = Eth1AddressSignerIdentifier.fromPublicKey(publicKey);
-    assertThat(signerIdentifier.validate(PublicKeyUtils.createKeyFrom("0xbb"))).isFalse();
+    assertThat(signerIdentifier.validate((ECPublicKey) secp256k1KeyPair2.getPublic())).isFalse();
   }
 
   @Test
   void validateFailsForNullPrimaryKey() {
-    final ECPublicKey publicKey = PublicKeyUtils.createKeyFrom("0xab");
+    final ECPublicKey publicKey = (ECPublicKey) secp256k1KeyPair.getPublic();
     final SignerIdentifier signerIdentifier = Eth1AddressSignerIdentifier.fromPublicKey(publicKey);
     assertThat(signerIdentifier.validate(null)).isFalse();
   }
 
   @Test
   void correctEth1AddressIsGeneratedFromPublicKey() {
-    final ECPublicKey publicKey = PublicKeyUtils.createKeyFrom("0xab");
+    final ECPublicKey publicKey = (ECPublicKey) secp256k1KeyPair.getPublic();
     final SignerIdentifier signerIdentifier = Eth1AddressSignerIdentifier.fromPublicKey(publicKey);
-    final String prefixRemovedAddress =
-        AddressUtil.remove0xPrefix(
-            Keys.getAddress(EthPublicKeyUtils.toHexString(publicKey)).toLowerCase(Locale.US));
-    assertThat(signerIdentifier.toStringIdentifier()).isEqualTo(prefixRemovedAddress);
+
+    // web3j.crypto.Keys.getAddress() returns lower case address without 0x prefix
+    final String expectedAddress =
+        Keys.getAddress(EthPublicKeyUtils.ecPublicKeyToBigInteger(publicKey));
+    assertThat(signerIdentifier.toStringIdentifier()).isEqualTo(expectedAddress);
+    assertThat(signerIdentifier.toStringIdentifier()).doesNotStartWithIgnoringCase("0x");
+    assertThat(signerIdentifier.toStringIdentifier()).isLowerCase();
+  }
+
+  public static String convertAlphabetsToUpperCase(final String input) {
+    final char[] chars = input.toCharArray();
+    int count = 0;
+
+    for (int i = 0; i < chars.length && count < 2; i++) {
```

Comment:
why do only the first two characters need to be uppercase?

Reply:
The intent was just to test that if mixed case hex are used as identifiers, they are normalized to be lower case. Since we are using random public keys, we are just converting the first two alpha characters to upper case.

Reply:
I've changed the method name and its access modifier to make the intent clear.